packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -70,20 +85,20 @@ class DatePicker extends React.Component {
     this.state = {
       locale: 'en-US', // TODO: Get the locale from i18n
       dateFormat: 'MM/DD/YYYY', // TODO: Get the locale from i18n
-      selectedDate: DateUtil.createSafeDate(props.selectedDate, 'MM/DD/YYYY'),
+      selectedDate: DateUtil.createSafeDate(props.selectedDate),
     };
 
     this.handleChange = this.handleChange.bind(this);
   }
 
-  handleChange(date) {
+  handleChange(date, event) {
     this.setState({
       selectedDate: date,
     });
 
     if (this.props.onChange) {
       const dateString = date && date.isValid() ? date.format(this.state.dateFormat) : '';
```

Comment:
I actually don't need to make this check because handleChange gets called from [here](https://github.com/Hacker0x01/react-datepicker/blob/master/src/datepicker.jsx#L234) only when an acceptable change has been made. So I've removed the check here.

https://github.com/cerner/terra-core/pull/452/commits/109a6590872e9de8cfee253af0a9fb3366ec60c6